src/components/common/TouchableTextIcon.js

```diff
@@ -0,0 +1,48 @@
+// @flow
+
+import * as React from 'react'
+import { StyleSheet, TouchableOpacity } from 'react-native'
+import MaterialIcons from 'react-native-vector-icons/MaterialIcons'
+
+import T from '../../modules/UI/components/FormattedText/FormattedText.ui.js'
+import { THEME } from '../../theme/variables/airbitz.js'
+import { scale } from '../../util/scaling.js'
+
+type Props = {
+  containerStyle?: StyleSheet.Styles,
+  icon?: React.Node,
+  onPress: Function,
+  title: string | React.Node
+}
+
+export class TouchableTextIcon extends React.Component<Props> {
+  render() {
+    const { containerStyle, icon, onPress, title } = this.props
+    return (
+      <TouchableOpacity onPress={onPress} style={containerStyle || styles.textIconContainer}>
+        {typeof title === 'string' ? (
+          <T style={styles.iconText} ellipsizeMode="middle" numberOfLines={1}>
+            {title}
+          </T>
+        ) : (
+          title
+        )}
+        {icon || <MaterialIcons name="keyboard-arrow-down" color={THEME.COLORS.WHITE} size={scale(25)} />}
+      </TouchableOpacity>
+    )
+  }
+}
+
+const styles = StyleSheet.create({
+  textIconContainer: {
+    position: 'relative',
+    flexDirection: 'row',
+    width: '100%',
```

Comment:
Please avoid `width` at all costs. Use `flex: 1` if you want to take up the available space. Flexbox is designed to avoid the need for fixed dimensions, since things can just expand and contract to fit instead.